### ISSUE TITLE
Missing deprecation notice for `ext_authz/allowed_headers`

### ISSUE THREAD

**Issue by phlax** [envoyproxy/envoy#25791]
As raised on slack if you set the now (since 1.25.0) deprecated `allowed_headers` field it results in a deprecation warning:

```console

[2023-01-23 06:09:44.065][6][warning][misc] [source/common/protobuf/message_validator_impl.cc:21] Deprecated field: type envoy.extensions.filters.http.ext_authz.v3.AuthorizationRequest Using deprecated option 'envoy.extensions.filters │
│ .http.ext_authz.v3.AuthorizationRequest.allowed_headers' from file ext_authz.proto. This configuration will be removed from Envoy soon. Please see https://www.envoyproxy.io/docs/envoy/latest/version_history/version_history for details │
│ . If continued use of this field is absolutely necessary, see https://www.envoyproxy.io/docs/envoy/latest/configuration/operations/runtime#using-runtime-overrides-for-deprecated-features for how to apply a temporary and highly discour │
│ aged override.

```

but it seems no deprecation was added to the changelog and the docs dont show the field as deprecated here - https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/http/ext_authz/v3/ext_authz.proto.html?highlight=envoy%20filters%20http%20ext_authz

**Comment by phlax** [comment]
wierd doesnt seem like anything has been deprecated afaict

**Comment by rulex123** [comment]
There's mention of that field being deprecated in the v1.25.0 changelog https://www.envoyproxy.io/docs/envoy/latest/version_history/v1.25/v1.25.0

From the `New features` section:

<img width="885" alt="Screenshot 2023-02-27 at 17 46 19" src="https://user-images.githubusercontent.com/29862113/221626496-f3aa2be3-078f-49bf-b792-687475f76e9e.png">

Do we need something more?

**Comment by phlax** [comment]
re the api - ideally we would auto generate something here but i think we dont so need a manual comment

**Comment by phlax** [comment]
Fixed in #25797
